source/Halibut.Tests/TcpClientCloseImmediatelyFixture.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Support;
+using Halibut.Tests.Transport.Streams;
+using Halibut.Tests.Util;
+using Halibut.Transport;
+using NUnit.Framework;
+
+namespace Halibut.Tests
+{
+    public class TcpClientCloseImmediatelyFixture : BaseTest
+    {
+        [Test]
+        public async Task DoesNotWaitNew()
+        {
+            var (client, clientStream, serverStream) = await BuildTcpClientAndTcpListener(CancellationToken);
+
+            await WriteToStreamUntilBlocked(clientStream);
+
+            Logger.Information("Writer has blocked, closing...");
+            var stopWatch = Stopwatch.StartNew();
+            client.CloseImmediately();
+            stopWatch.Stop();
+            Logger.Information($"Close completed, duration: {stopWatch.Elapsed.TotalMilliseconds}ms");
+            
+            stopWatch.ElapsedMilliseconds.Should().BeLessThan(10);
+
+            byte[] received = new byte[65536];
+            await AssertAsync.Throws<IOException>(async () => _ = await serverStream.ReadAsync(received, 0, received.Length, CancellationToken));
```

Comment:
It would be worth documenting why the IOException is meaningful to `CloseImmediatly`
Maybe add:
```
// Close immediately will result in the buffers on the senders side being dropped and a TCP RST being sent,
// Its hard to see the data in the senders bufferes being dropped however we can observe a TCP RST is recieved
// by the reciever. So we assert on that.
// An ordinary close will not result in the reciever recieving a TCP RST.
```

It would make sense to assert the exception is one that talks about TCP RST.

Don't forget that the `ReadAsync` doesn't need to fill the buffer given it so `ReadAsync` ought to be in a loop until it returns a value of zero.